lib/genre.rb

```diff
@@ -0,0 +1,24 @@
+class Genre
+  include Concerns
+
+  attr_accessor :name, :artist, :genre, :songs
+  @@all = []
+
+  def initialize(name)
+    @name = name
+    @songs = []
+    @artist = []
+  end
+
+  def add_song(song)
+    song.genre ||= self.name
+    @songs << song unless @songs.include? song
```

Comment:
Trainer's feedback: This is similar to add_song method in artist class. Combine this in a single method using a module.